naples/routes/auth.py

```diff
@@ -51,12 +56,25 @@ def login(
     if not user:
         log(log.ERROR, "User [%s] wrong username (email) or password", form_data.username)
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Invalid credentials")
+
     # admin user can not get API token
     if user.role == s.UserRole.ADMIN.value:
         log(log.ERROR, "User [%s] is an admin user", user.email)
         raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Admin user can not get API token")
     log(log.INFO, "User [%s] logged in", user.email)
 
+    # update last subscription in db with stripe data every 3 days
+    if user.subscription and user.subscription.last_checked_date < datetime.now() - timedelta(
+        days=CFG.DAYS_BEFORE_UPDATE
+    ):
+        stripe_subscription_data = stripe.Subscription.retrieve(user.subscription.subscription_stripe_id)
+
+        product = get_product_by_id(stripe_subscription_data["items"]["data"][0]["plan"]["id"], db)
+
+        save_state_subscription_from_stripe(user.subscription, product, db)
+
+        log(log.INFO, "Subscription state updated for user [%s]", user.email)
+
```

Comment:
Please do not forget to check if this subscription is expired before returning token to user